Failed to build TTK w/ 64-bit IdType

**Bug**
Compiler error with `TTK_ENABLE_64BIT_IDS=ON` 

**To Reproduce**
Steps to reproduce the behavior:
1.
```
     ccmake \
       -GNinja \
       -S~/Downloads/ttk \
       -DCMAKE_C_COMPILER=/usr/bin/clang \
       -DCMAKE_CXX_COMPILER=/usr/bin/clang++ \
       -DCMAKE_INSTALL_PREFIX=~/installs/ttk-dev-release  \
       -DTTK_ENABLE_64BIT_IDS=ON
```
2. `ninja`
3.  See an error

```
[jaswantp@jaswantp-arch ttk-dev-release]$ ninja
[102/1865] Building CXX object core/vtk/ttkArrayPreconditioning/CMakeFiles/ttkArrayPreconditioning.dir/ttkArrayPreconditioning.cpp.o
FAILED: core/vtk/ttkArrayPreconditioning/CMakeFiles/ttkArrayPreconditioning.dir/ttkArrayPreconditioning.cpp.o 
/usr/bin/clang++ -DPARAVIEW_VERSION_MAJOR=5 -DPARAVIEW_VERSION_MINOR=9 -DTTK_CELL_ARRAY_NEW -DTTK_ENABLE_64BIT_IDS -DTTK_ENABLE_OPENMP -DTTK_ENABLE_SCIKIT_LEARN -DTTK_ENABLE_SHARED_BASE_LIBRARIES -DTTK_ENABLE_SQLITE3 -DttkArrayPreconditioning_EXPORTS -Icore/vtk/ttkArrayPreconditioning -I/home/jaswantp/Downloads/ttk/core/vtk/ttkArrayPreconditioning -Icore/vtk/ttkAlgorithm -I/home/jaswantp/Downloads/ttk/core/vtk/ttkAlgorithm -I/home/jaswantp/Downloads/ttk/core/base/triangulation -I/home/jaswantp/Downloads/ttk/core/base/abstractTriangulation -I/home/jaswantp/Downloads/ttk/core/base/common -I/home/jaswantp/Downloads/ttk/core/base/geometry -I/home/jaswantp/Downloads/ttk/core/base/explicitTriangulation -I/home/jaswantp/Downloads/ttk/core/base/skeleton -Icore/base/skeleton -I/home/jaswantp/Downloads/ttk/core/base/implicitTriangulation -I/home/jaswantp/Downloads/ttk/core/base/periodicImplicitTriangulation -isystem /usr/include/python3.9 -isystem /home/jaswantp/installs/pv-5.9rc4-release/include/paraview-5.9 -O3 -DNDEBUG -fPIC -Wall -Wshadow -march=native -Wfatal-errors -fopenmp=libomp -std=gnu++11 -MD -MT core/vtk/ttkArrayPreconditioning/CMakeFiles/ttkArrayPreconditioning.dir/ttkArrayPreconditioning.cpp.o -MF core/vtk/ttkArrayPreconditioning/CMakeFiles/ttkArrayPreconditioning.dir/ttkArrayPreconditioning.cpp.o.d -o core/vtk/ttkArrayPreconditioning/CMakeFiles/ttkArrayPreconditioning.dir/ttkArrayPreconditioning.cpp.o -c /home/jaswantp/Downloads/ttk/core/vtk/ttkArrayPreconditioning/ttkArrayPreconditioning.cpp
In file included from /home/jaswantp/Downloads/ttk/core/vtk/ttkArrayPreconditioning/ttkArrayPreconditioning.cpp:11:
/home/jaswantp/Downloads/ttk/core/vtk/ttkAlgorithm/ttkMacros.h:8:31: fatal error: unknown type name 'vtkLongLongArray'
using ttkSimplexIdTypeArray = vtkLongLongArray;
                              ^
1 error generated.
[107/1865] Building CXX object core/vtk/ttkAlgorithm/CMakeFiles/ttkAlgorithm.dir/ttkAlgorithm.cpp.o
FAILED: core/vtk/ttkAlgorithm/CMakeFiles/ttkAlgorithm.dir/ttkAlgorithm.cpp.o 
/usr/bin/clang++ -DPARAVIEW_VERSION_MAJOR=5 -DPARAVIEW_VERSION_MINOR=9 -DTTK_CELL_ARRAY_NEW -DTTK_ENABLE_64BIT_IDS -DTTK_ENABLE_OPENMP -DTTK_ENABLE_SCIKIT_LEARN -DTTK_ENABLE_SHARED_BASE_LIBRARIES -DTTK_ENABLE_SQLITE3 -DttkAlgorithm_EXPORTS -Icore/vtk/ttkAlgorithm -I/home/jaswantp/Downloads/ttk/core/vtk/ttkAlgorithm -I/home/jaswantp/Downloads/ttk/core/base/triangulation -I/home/jaswantp/Downloads/ttk/core/base/abstractTriangulation -I/home/jaswantp/Downloads/ttk/core/base/common -I/home/jaswantp/Downloads/ttk/core/base/geometry -I/home/jaswantp/Downloads/ttk/core/base/explicitTriangulation -I/home/jaswantp/Downloads/ttk/core/base/skeleton -Icore/base/skeleton -I/home/jaswantp/Downloads/ttk/core/base/implicitTriangulation -I/home/jaswantp/Downloads/ttk/core/base/periodicImplicitTriangulation -isystem /usr/include/python3.9 -isystem /home/jaswantp/installs/pv-5.9rc4-release/include/paraview-5.9 -O3 -DNDEBUG -fPIC -Wall -Wshadow -march=native -Wfatal-errors -fopenmp=libomp -std=gnu++11 -MD -MT core/vtk/ttkAlgorithm/CMakeFiles/ttkAlgorithm.dir/ttkAlgorithm.cpp.o -MF core/vtk/ttkAlgorithm/CMakeFiles/ttkAlgorithm.dir/ttkAlgorithm.cpp.o.d -o core/vtk/ttkAlgorithm/CMakeFiles/ttkAlgorithm.dir/ttkAlgorithm.cpp.o -c /home/jaswantp/Downloads/ttk/core/vtk/ttkAlgorithm/ttkAlgorithm.cpp
In file included from /home/jaswantp/Downloads/ttk/core/vtk/ttkAlgorithm/ttkAlgorithm.cpp:2:
/home/jaswantp/Downloads/ttk/core/vtk/ttkAlgorithm/ttkMacros.h:8:31: fatal error: unknown type name 'vtkLongLongArray'
using ttkSimplexIdTypeArray = vtkLongLongArray;
                              ^
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.      Program arguments: /usr/bin/clang++ -DPARAVIEW_VERSION_MAJOR=5 -DPARAVIEW_VERSION_MINOR=9 -DTTK_CELL_ARRAY_NEW -DTTK_ENABLE_64BIT_IDS -DTTK_ENABLE_OPENMP -DTTK_ENABLE_SCIKIT_LEARN -DTTK_ENABLE_SHARED_BASE_LIBRARIES -DTTK_ENABLE_SQLITE3 -DttkAlgorithm_EXPORTS -Icore/vtk/ttkAlgorithm -I/home/jaswantp/Downloads/ttk/core/vtk/ttkAlgorithm -I/home/jaswantp/Downloads/ttk/core/base/triangulation -I/home/jaswantp/Downloads/ttk/core/base/abstractTriangulation -I/home/jaswantp/Downloads/ttk/core/base/common -I/home/jaswantp/Downloads/ttk/core/base/geometry -I/home/jaswantp/Downloads/ttk/core/base/explicitTriangulation -I/home/jaswantp/Downloads/ttk/core/base/skeleton -Icore/base/skeleton -I/home/jaswantp/Downloads/ttk/core/base/implicitTriangulation -I/home/jaswantp/Downloads/ttk/core/base/periodicImplicitTriangulation -isystem /usr/include/python3.9 -isystem /home/jaswantp/installs/pv-5.9rc4-release/include/paraview-5.9 -O3 -DNDEBUG -fPIC -Wall -Wshadow -march=native -Wfatal-errors -fopenmp=libomp -std=gnu++11 -MD -MT core/vtk/ttkAlgorithm/CMakeFiles/ttkAlgorithm.dir/ttkAlgorithm.cpp.o -MF core/vtk/ttkAlgorithm/CMakeFiles/ttkAlgorithm.dir/ttkAlgorithm.cpp.o.d -o core/vtk/ttkAlgorithm/CMakeFiles/ttkAlgorithm.dir/ttkAlgorithm.cpp.o -c /home/jaswantp/Downloads/ttk/core/vtk/ttkAlgorithm/ttkAlgorithm.cpp 
1.      <eof> parser at end of file
 #0 0x00007f982d0b915b llvm::sys::PrintStackTrace(llvm::raw_ostream&) (/usr/bin/../lib/libLLVM-11.so+0xa7615b)
 #1 0x00007f982d0b6e84 llvm::sys::RunSignalHandlers() (/usr/bin/../lib/libLLVM-11.so+0xa73e84)
 #2 0x00007f982cfc96a9 (/usr/bin/../lib/libLLVM-11.so+0x9866a9)
 #3 0x00007f982c2da6a0 __restore_rt (/usr/bin/../lib/libc.so.6+0x3d6a0)
 #4 0x00007f98328867ec clang::ImportError::convertToErrorCode() const (/usr/bin/../lib/libclang-cpp.so.11+0x7e47ec)
 #5 0x00007f98335be65b clang::Sema::LookupDestructor(clang::CXXRecordDecl*) (/usr/bin/../lib/libclang-cpp.so.11+0x151c65b)
 #6 0x00007f98330d24f9 (/usr/bin/../lib/libclang-cpp.so.11+0x10304f9)
 #7 0x00007f98330d3b42 (/usr/bin/../lib/libclang-cpp.so.11+0x1031b42)
 #8 0x00007f98330d3b42 (/usr/bin/../lib/libclang-cpp.so.11+0x1031b42)
 #9 0x00007f98330d3b42 (/usr/bin/../lib/libclang-cpp.so.11+0x1031b42)
#10 0x00007f98330d3626 (/usr/bin/../lib/libclang-cpp.so.11+0x1031626)
#11 0x00007f98330d3d72 (/usr/bin/../lib/libclang-cpp.so.11+0x1031d72)
#12 0x00007f98330d41c6 (/usr/bin/../lib/libclang-cpp.so.11+0x10321c6)
#13 0x00007f98330d4ab4 clang::Sema::ActOnEndOfTranslationUnit() (/usr/bin/../lib/libclang-cpp.so.11+0x1032ab4)
#14 0x00007f9832ad08a3 clang::Parser::ParseTopLevelDecl(clang::OpaquePtr<clang::DeclGroupRef>&, bool) (/usr/bin/../lib/libclang-cpp.so.11+0xa2e8a3)
#15 0x00007f9832a00c9a clang::ParseAST(clang::Sema&, bool, bool) (/usr/bin/../lib/libclang-cpp.so.11+0x95ec9a)
#16 0x00007f9834385c89 clang::FrontendAction::Execute() (/usr/bin/../lib/libclang-cpp.so.11+0x22e3c89)
#17 0x00007f98343356ce clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/usr/bin/../lib/libclang-cpp.so.11+0x22936ce)
#18 0x00007f98344035cd clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/usr/bin/../lib/libclang-cpp.so.11+0x23615cd)
#19 0x00005618eb76f852 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/usr/bin/clang+++0x12852)
#20 0x00005618eb76cdfc (/usr/bin/clang+++0xfdfc)
#21 0x00007f9834000a65 (/usr/bin/../lib/libclang-cpp.so.11+0x1f5ea65)
#22 0x00007f982cfc97c3 llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/usr/bin/../lib/libLLVM-11.so+0x9867c3)
#23 0x00007f983400170e (/usr/bin/../lib/libclang-cpp.so.11+0x1f5f70e)
#24 0x00007f9833fd6f78 clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&) const (/usr/bin/../lib/libclang-cpp.so.11+0x1f34f78)
#25 0x00007f9833fd7867 clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*> >&) const (/usr/bin/../lib/libclang-cpp.so.11+0x1f35867)
#26 0x00007f9833fe334c clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*> >&) (/usr/bin/../lib/libclang-cpp.so.11+0x1f4134c)
#27 0x00005618eb76ac02 main (/usr/bin/clang+++0xdc02)
#28 0x00007f982c2c5152 __libc_start_main (/usr/bin/../lib/libc.so.6+0x28152)
#29 0x00005618eb76c76e _start (/usr/bin/clang+++0xf76e)
clang-11: fatal error: clang frontend command failed due to signal (use -v to see invocation)
clang version 11.0.1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
clang-11: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang-11: note: diagnostic msg: /tmp/ttkAlgorithm-45e995.cpp
clang-11: note: diagnostic msg: /tmp/ttkAlgorithm-45e995.sh
clang-11: note: diagnostic msg: 

********************
[109/1865] Building CXX object core/vtk/ttkTriangulationRequest/CMakeFiles/ttkTriangulationRequest.dir/ttkTriangulationRequest.cpp.o
FAILED: core/vtk/ttkTriangulationRequest/CMakeFiles/ttkTriangulationRequest.dir/ttkTriangulationRequest.cpp.o 
/usr/bin/clang++ -DPARAVIEW_VERSION_MAJOR=5 -DPARAVIEW_VERSION_MINOR=9 -DTTK_CELL_ARRAY_NEW -DTTK_ENABLE_64BIT_IDS -DTTK_ENABLE_OPENMP -DTTK_ENABLE_SCIKIT_LEARN -DTTK_ENABLE_SHARED_BASE_LIBRARIES -DTTK_ENABLE_SQLITE3 -DttkTriangulationRequest_EXPORTS -Icore/vtk/ttkTriangulationRequest -I/home/jaswantp/Downloads/ttk/core/vtk/ttkTriangulationRequest -Icore/vtk/ttkAlgorithm -I/home/jaswantp/Downloads/ttk/core/vtk/ttkAlgorithm -I/home/jaswantp/Downloads/ttk/core/base/triangulation -I/home/jaswantp/Downloads/ttk/core/base/abstractTriangulation -I/home/jaswantp/Downloads/ttk/core/base/common -I/home/jaswantp/Downloads/ttk/core/base/geometry -I/home/jaswantp/Downloads/ttk/core/base/explicitTriangulation -I/home/jaswantp/Downloads/ttk/core/base/skeleton -Icore/base/skeleton -I/home/jaswantp/Downloads/ttk/core/base/implicitTriangulation -I/home/jaswantp/Downloads/ttk/core/base/periodicImplicitTriangulation -isystem /usr/include/python3.9 -isystem /home/jaswantp/installs/pv-5.9rc4-release/include/paraview-5.9 -O3 -DNDEBUG -fPIC -Wall -Wshadow -march=native -Wfatal-errors -fopenmp=libomp -std=gnu++11 -MD -MT core/vtk/ttkTriangulationRequest/CMakeFiles/ttkTriangulationRequest.dir/ttkTriangulationRequest.cpp.o -MF core/vtk/ttkTriangulationRequest/CMakeFiles/ttkTriangulationRequest.dir/ttkTriangulationRequest.cpp.o.d -o core/vtk/ttkTriangulationRequest/CMakeFiles/ttkTriangulationRequest.dir/ttkTriangulationRequest.cpp.o -c /home/jaswantp/Downloads/ttk/core/vtk/ttkTriangulationRequest/ttkTriangulationRequest.cpp
In file included from /home/jaswantp/Downloads/ttk/core/vtk/ttkTriangulationRequest/ttkTriangulationRequest.cpp:1:
/home/jaswantp/Downloads/ttk/core/vtk/ttkAlgorithm/ttkMacros.h:8:31: fatal error: unknown type name 'vtkLongLongArray'
using ttkSimplexIdTypeArray = vtkLongLongArray;
                              ^
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.      Program arguments: /usr/bin/clang++ -DPARAVIEW_VERSION_MAJOR=5 -DPARAVIEW_VERSION_MINOR=9 -DTTK_CELL_ARRAY_NEW -DTTK_ENABLE_64BIT_IDS -DTTK_ENABLE_OPENMP -DTTK_ENABLE_SCIKIT_LEARN -DTTK_ENABLE_SHARED_BASE_LIBRARIES -DTTK_ENABLE_SQLITE3 -DttkTriangulationRequest_EXPORTS -Icore/vtk/ttkTriangulationRequest -I/home/jaswantp/Downloads/ttk/core/vtk/ttkTriangulationRequest -Icore/vtk/ttkAlgorithm -I/home/jaswantp/Downloads/ttk/core/vtk/ttkAlgorithm -I/home/jaswantp/Downloads/ttk/core/base/triangulation -I/home/jaswantp/Downloads/ttk/core/base/abstractTriangulation -I/home/jaswantp/Downloads/ttk/core/base/common -I/home/jaswantp/Downloads/ttk/core/base/geometry -I/home/jaswantp/Downloads/ttk/core/base/explicitTriangulation -I/home/jaswantp/Downloads/ttk/core/base/skeleton -Icore/base/skeleton -I/home/jaswantp/Downloads/ttk/core/base/implicitTriangulation -I/home/jaswantp/Downloads/ttk/core/base/periodicImplicitTriangulation -isystem /usr/include/python3.9 -isystem /home/jaswantp/installs/pv-5.9rc4-release/include/paraview-5.9 -O3 -DNDEBUG -fPIC -Wall -Wshadow -march=native -Wfatal-errors -fopenmp=libomp -std=gnu++11 -MD -MT core/vtk/ttkTriangulationRequest/CMakeFiles/ttkTriangulationRequest.dir/ttkTriangulationRequest.cpp.o -MF core/vtk/ttkTriangulationRequest/CMakeFiles/ttkTriangulationRequest.dir/ttkTriangulationRequest.cpp.o.d -o core/vtk/ttkTriangulationRequest/CMakeFiles/ttkTriangulationRequest.dir/ttkTriangulationRequest.cpp.o -c /home/jaswantp/Downloads/ttk/core/vtk/ttkTriangulationRequest/ttkTriangulationRequest.cpp 
1.      <eof> parser at end of file
 #0 0x00007f625eec215b llvm::sys::PrintStackTrace(llvm::raw_ostream&) (/usr/bin/../lib/libLLVM-11.so+0xa7615b)
 #1 0x00007f625eebfe84 llvm::sys::RunSignalHandlers() (/usr/bin/../lib/libLLVM-11.so+0xa73e84)
 #2 0x00007f625edd26a9 (/usr/bin/../lib/libLLVM-11.so+0x9866a9)
 #3 0x00007f625e0e36a0 __restore_rt (/usr/bin/../lib/libc.so.6+0x3d6a0)
 #4 0x00007f626468f7ec clang::ImportError::convertToErrorCode() const (/usr/bin/../lib/libclang-cpp.so.11+0x7e47ec)
 #5 0x00007f62653c765b clang::Sema::LookupDestructor(clang::CXXRecordDecl*) (/usr/bin/../lib/libclang-cpp.so.11+0x151c65b)
 #6 0x00007f6264edb4f9 (/usr/bin/../lib/libclang-cpp.so.11+0x10304f9)
 #7 0x00007f6264edcb42 (/usr/bin/../lib/libclang-cpp.so.11+0x1031b42)
 #8 0x00007f6264edcb42 (/usr/bin/../lib/libclang-cpp.so.11+0x1031b42)
 #9 0x00007f6264edcb42 (/usr/bin/../lib/libclang-cpp.so.11+0x1031b42)
#10 0x00007f6264edc626 (/usr/bin/../lib/libclang-cpp.so.11+0x1031626)
#11 0x00007f6264edcd72 (/usr/bin/../lib/libclang-cpp.so.11+0x1031d72)
#12 0x00007f6264edd1c6 (/usr/bin/../lib/libclang-cpp.so.11+0x10321c6)
#13 0x00007f6264eddab4 clang::Sema::ActOnEndOfTranslationUnit() (/usr/bin/../lib/libclang-cpp.so.11+0x1032ab4)
#14 0x00007f62648d98a3 clang::Parser::ParseTopLevelDecl(clang::OpaquePtr<clang::DeclGroupRef>&, bool) (/usr/bin/../lib/libclang-cpp.so.11+0xa2e8a3)
#15 0x00007f6264809c9a clang::ParseAST(clang::Sema&, bool, bool) (/usr/bin/../lib/libclang-cpp.so.11+0x95ec9a)
#16 0x00007f626618ec89 clang::FrontendAction::Execute() (/usr/bin/../lib/libclang-cpp.so.11+0x22e3c89)
#17 0x00007f626613e6ce clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/usr/bin/../lib/libclang-cpp.so.11+0x22936ce)
#18 0x00007f626620c5cd clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/usr/bin/../lib/libclang-cpp.so.11+0x23615cd)
#19 0x000055e9d48a2852 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/usr/bin/clang+++0x12852)
#20 0x000055e9d489fdfc (/usr/bin/clang+++0xfdfc)
#21 0x00007f6265e09a65 (/usr/bin/../lib/libclang-cpp.so.11+0x1f5ea65)
#22 0x00007f625edd27c3 llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/usr/bin/../lib/libLLVM-11.so+0x9867c3)
#23 0x00007f6265e0a70e (/usr/bin/../lib/libclang-cpp.so.11+0x1f5f70e)
#24 0x00007f6265ddff78 clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&) const (/usr/bin/../lib/libclang-cpp.so.11+0x1f34f78)
#25 0x00007f6265de0867 clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*> >&) const (/usr/bin/../lib/libclang-cpp.so.11+0x1f35867)
#26 0x00007f6265dec34c clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*> >&) (/usr/bin/../lib/libclang-cpp.so.11+0x1f4134c)
#27 0x000055e9d489dc02 main (/usr/bin/clang+++0xdc02)
#28 0x00007f625e0ce152 __libc_start_main (/usr/bin/../lib/libc.so.6+0x28152)
#29 0x000055e9d489f76e _start (/usr/bin/clang+++0xf76e)
clang-11: fatal error: clang frontend command failed due to signal (use -v to see invocation)
clang version 11.0.1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
clang-11: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang-11: note: diagnostic msg: /tmp/ttkTriangulationRequest-ac0593.cpp
clang-11: note: diagnostic msg: /tmp/ttkTriangulationRequest-ac0593.sh
clang-11: note: diagnostic msg: 

********************
[115/1865] Building CXX object core/vtk/ttkTrackingFromOverlap/CMakeFiles/ttkTrackingFromOverlap.dir/ttkTrackingFromOverlap.cpp.o
ninja: build stopped: subcommand failed.
```
**Screenshots**
CMake configuration, summary and error.
![image](https://user-images.githubusercontent.com/40538987/105527342-945ace80-5d09-11eb-8e30-187e8bb2868e.png)


**System (please complete the following information):**
 - OS: Arch Linux 5.10.9 x86_64
 - Compiler clang++ 11.0.1
 - Installed TTK's optional dependencies (Python 3.9, Paraview v5.9.0-RC4 (local build))

**Additional context**
If I turn that flag off, it builds fine.

**What I tried**
From the first error, it looks like [`ttkMacros.h`](https://github.com/topology-tool-kit/ttk/blob/dev/core/vtk/ttkAlgorithm/ttkMacros.h) is missing `#include <vtkLongLongArray.h>`. So I figured, ehh, that's about it and went ahead to _include_ it in there like so.

![image](https://user-images.githubusercontent.com/40538987/105527940-57430c00-5d0a-11eb-815d-80ad030cd282.png)

Not enough, there's another error now.
```
[jaswantp@jaswantp-arch ttk-dev-release]$ ninja
[179/1865] Building CXX object core/vtk/ttkTopologicalSimplification/CMakeFiles/ttkTopologicalSimplification.dir/ttkTopologicalSimplification.cpp.o
FAILED: core/vtk/ttkTopologicalSimplification/CMakeFiles/ttkTopologicalSimplification.dir/ttkTopologicalSimplification.cpp.o 
/usr/bin/clang++ -DBOOST_ALL_NO_LIB -DPARAVIEW_VERSION_MAJOR=5 -DPARAVIEW_VERSION_MINOR=9 -DTTK_CELL_ARRAY_NEW -DTTK_ENABLE_64BIT_IDS -DTTK_ENABLE_OPENMP -DTTK_ENABLE_SCIKIT_LEARN -DTTK_ENABLE_SHARED_BASE_LIBRARIES -DTTK_ENABLE_SQLITE3 -DttkTopologicalSimplification_EXPORTn -Icore/vtk/ttkAlgorithm -I/home/jaswantp/Downloads/ttk/core/vtk/ttkAlgorithm -I/home/jaswantp/Downloads/ttk/core/base/triangulation -I/home/jaswantp/Downloads/ttk/core/base/abstractTriangulation -I/home/jaswantp/Downloads/ttk/core/base/common -I/home/jaswantp/Downloads/tds/ttk/core/base/skeleton -Icore/base/skeleton -I/home/jaswantp/Downloads/ttk/core/base/implicitTriangulation -I/home/jaswantp/Downloads/ttk/core/base/periodicImplicitTriangulation -I/home/jaswantp/Downloads/ttk/core/base/topologicalSimplification -I/home/jaswantp/Downloadstalls/pv-5.9rc4-release/include/paraview-5.9 -O3 -DNDEBUG -fPIC -Wall -Wshadow -march=native -Wfatal-errors -fopenmp=libomp -std=gnu++11 -MD -MT core/vtk/ttkTopologicalSimplification/CMakeFiles/ttkTopologicalSimplification.dir/ttkTopologicalSimplification.cpp.o -MF core/vo.d -o core/vtk/ttkTopologicalSimplification/CMakeFiles/ttkTopologicalSimplification.dir/ttkTopologicalSimplification.cpp.o -c /home/jaswantp/Downloads/ttk/core/vtk/ttkTopologicalSimplification/ttkTopologicalSimplification.cpp
/home/jaswantp/Downloads/ttk/core/vtk/ttkTopologicalSimplification/ttkTopologicalSimplification.cpp:168:21: fatal error: no matching member function for call to 'execute'
        ret = this->execute(
              ~~~~~~^~~~~~~
/home/jaswantp/installs/pv-5.9rc4-release/include/paraview-5.9/vtkSetGet.h:929:44: note: expanded from macro 'vtkTemplateMacro'
  vtkTemplateMacroCase(VTK_DOUBLE, double, call);                                                  \
                                           ^~~~
/home/jaswantp/installs/pv-5.9rc4-release/include/paraview-5.9/vtkSetGet.h:925:5: note: expanded from macro 'vtkTemplateMacroCase'
    call;                                                                                          \
    ^~~~
/home/jaswantp/Downloads/ttk/core/base/topologicalSimplification/TopologicalSimplification.h:259:37: note: candidate function template not viable: no known conversion from 'int *' to 'const ttk::SimplexId *const' (aka 'const long long *const') for 3rd argument
int ttk::TopologicalSimplification::execute(
                                    ^
1 error generated.
```
Same error when I tried to compile TTK with `gcc` against a `clang` built paraview (both with 64-bit IDS enabled)

It could be something to do with my local build of paraview or, it somehow went unnoticed for quite a long time.  None of the Azure pipelines has `TTK_ENABLE_64_BIT_IDS=ON`.

I guess this will resolve itself once the [move](https://github.com/topology-tool-kit/ttk/pull/318) to VTK Range-based DataArray API is fully complete. I'd be glad to help achieve that if it needs more manpower.

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -117,6 +117,7 @@ jobs:
       matrix:
         kamikaze: [KAMIKAZE=TRUE, KAMIKAZE=FALSE]
         omp: [OPENMP=TRUE, OPENMP=FALSE]
+        simplexId: [64BIT_IDS=TRUE, 64BIT_IDS=FALSE]
     steps:
     - uses: actions/checkout@v2
 
@@ -155,6 +156,7 @@ jobs:
           -DTTK_BUILD_STANDALONE_APPS=TRUE \
           -DTTK_ENABLE_${{ matrix.kamikaze }} \
           -DTTK_ENABLE_${{ matrix.omp }} \
+          -DTTK_ENABLE_${{ matrix.simplexId }} \
           $GITHUB_WORKSPACE
 
     - name: Use clang-check for compiler warnings

--- a/core/base/contourForests/ContourForestsTemplate.h
+++ b/core/base/contourForests/ContourForestsTemplate.h
@@ -315,7 +315,7 @@ namespace ttk {
         std::tuple<std::vector<SimplexId>, std::vector<SimplexId>> overlaps
           = getOverlaps(i);
         const SimplexId &partitionSize
-          = abs(std::get<0>(rangeJT) - std::get<1>(rangeJT))
+          = std::abs(std::get<0>(rangeJT) - std::get<1>(rangeJT))
             + std::get<0>(overlaps).size() + std::get<1>(overlaps).size();
 
         // ---------------

--- a/core/base/contourForestsTree/MergeTreeTemplate.h
+++ b/core/base/contourForestsTree/MergeTreeTemplate.h
@@ -875,12 +875,13 @@ namespace ttk {
           msg << "partition : " << static_cast<unsigned>(treeData_.partition);
           msg << ", isJT : " << isJT;
           msg << ",  size : ";
-          msg << "before  : " << abs(beforeEnd - beforeStart);
-          msg << ", main : " << abs(mainEnd - mainStart);
-          msg << ", after : " << abs(afterEnd - afterStart);
+          msg << "before  : " << std::abs(beforeEnd - beforeStart);
+          msg << ", main : " << std::abs(mainEnd - mainStart);
+          msg << ", after : " << std::abs(afterEnd - afterStart);
           msg << ", Total : ";
-          msg << abs(beforeEnd - beforeStart) + abs(mainEnd - mainStart)
-                   + abs(afterEnd - afterStart);
+          msg << std::abs(beforeEnd - beforeStart)
+                   + std::abs(mainEnd - mainStart)
+                   + std::abs(afterEnd - afterStart);
           this->printMsg(msg.str());
         }
       }

--- a/core/base/topologicalCompression/PersistenceDiagramCompression.h
+++ b/core/base/topologicalCompression/PersistenceDiagramCompression.h
@@ -239,7 +239,7 @@ int ttk::TopologicalCompression::PerformSimplification(
   const triangulationType &triangulation) {
 
   std::vector<int> inputOffsets(vertexNumber);
-  std::vector<int> critConstraints(nbConstraints);
+  std::vector<SimplexId> critConstraints(nbConstraints);
   std::vector<double> inArray(vertexNumber);
   // std::vector<int> *oo = new std::vector<int>(vertexNumber);
   decompressedOffsets_.resize(vertexNumber); // oo->data();
@@ -449,7 +449,7 @@ int ttk::TopologicalCompression::compressForPersistenceDiagram(
 
   const char *sq = SQMethod.c_str();
   int nbCrit = 0;
-  std::vector<int> simplifiedConstraints;
+  std::vector<SimplexId> simplifiedConstraints;
   if(strcmp(sq, "") == 0 && !ZFPOnly) {
     // No SQ: perform topological control
 

--- a/core/base/topologicalCompression/TopologicalCompression.h
+++ b/core/base/topologicalCompression/TopologicalCompression.h
@@ -166,10 +166,10 @@ namespace ttk {
     inline std::vector<double> &getDecompressedData() {
       return decompressedData_;
     }
-    inline std::vector<int> &getDecompressedOffsets() {
+    inline std::vector<SimplexId> &getDecompressedOffsets() {
       return decompressedOffsets_;
     }
-    inline std::vector<int> &getCompressedOffsets() {
+    inline std::vector<SimplexId> &getCompressedOffsets() {
       return compressedOffsets_;
     }
 
@@ -403,8 +403,8 @@ namespace ttk {
     int NbSegments{0};
     int rawFileLength{0};
     std::vector<double> decompressedData_{};
-    std::vector<int> decompressedOffsets_{};
-    std::vector<int> compressedOffsets_{};
+    std::vector<SimplexId> decompressedOffsets_{};
+    std::vector<SimplexId> compressedOffsets_{};
     int vertexNumberRead_{};
     char *fileName{};
 

--- a/core/vtk/ttkAlgorithm/ttkMacros.h
+++ b/core/vtk/ttkAlgorithm/ttkMacros.h
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <vtkIntArray.h>
+#include <vtkLongLongArray.h>
 
 #define TTK_COMMA ,
 

--- a/core/vtk/ttkJacobiSet/ttkJacobiSet.cpp
+++ b/core/vtk/ttkJacobiSet/ttkJacobiSet.cpp
@@ -123,7 +123,7 @@ int ttkJacobiSet::RequestData(vtkInformation *request,
   for(size_t i = 0; i < jacobiSet_.size(); i++) {
 
     int edgeId = jacobiSet_[i].first;
-    int vertexId0 = -1, vertexId1 = -1;
+    ttk::SimplexId vertexId0 = -1, vertexId1 = -1;
     triangulation->getEdgeVertex(edgeId, 0, vertexId0);
     triangulation->getEdgeVertex(edgeId, 1, vertexId1);
 
@@ -176,7 +176,7 @@ int ttkJacobiSet::RequestData(vtkInformation *request,
 
       for(size_t j = 0; j < jacobiSet_.size(); j++) {
         int edgeId = jacobiSet_[j].first;
-        int vertexId0 = -1, vertexId1 = -1;
+        ttk::SimplexId vertexId0 = -1, vertexId1 = -1;
         triangulation->getEdgeVertex(edgeId, 0, vertexId0);
         triangulation->getEdgeVertex(edgeId, 1, vertexId1);
 

--- a/core/vtk/ttkReebSpace/ttkReebSpace.cpp
+++ b/core/vtk/ttkReebSpace/ttkReebSpace.cpp
@@ -312,7 +312,7 @@ int ttkReebSpace::RequestData(vtkInformation *request,
 
       if((sheet) && (!sheet->pruned_)) {
 
-        int vertexId0 = -1, vertexId1 = -1;
+        ttk::SimplexId vertexId0 = -1, vertexId1 = -1;
         triangulation->getEdgeVertex(i, 0, vertexId0);
         triangulation->getEdgeVertex(i, 1, vertexId1);
 

--- a/core/vtk/ttkTopologicalSimplification/ttkTopologicalSimplification.cpp
+++ b/core/vtk/ttkTopologicalSimplification/ttkTopologicalSimplification.cpp
@@ -168,7 +168,7 @@ int ttkTopologicalSimplification::RequestData(
         ret = this->execute(
           static_cast<VTK_TT *>(ttkUtils::GetVoidPointer(inputScalars)),
           static_cast<VTK_TT *>(ttkUtils::GetVoidPointer(outputScalars)),
-          static_cast<int *>(ttkUtils::GetVoidPointer(identifiers)),
+          static_cast<SimplexId *>(ttkUtils::GetVoidPointer(identifiers)),
           static_cast<SimplexId *>(ttkUtils::GetVoidPointer(offsets)),
           static_cast<SimplexId *>(ttkUtils::GetVoidPointer(outputOffsets)),
           numberOfConstraints, *triangulation->getData()));

